src/java/org/apache/cassandra/auth/CassandraRoleManager.java

```diff
@@ -732,4 +737,32 @@ public Supplier<Map<RoleResource, Set<Role>>> bulkLoader()
             return entries;
         };
     }
+
+    protected void scheduleDisconnectInvalidRoleTask()
+    {
+        // Delay between executions, with jitter
+        int period = (int) DatabaseDescriptor.getInvalidClientDisconnectPeriod(TimeUnit.MILLISECONDS);
+        int jitter = (int) DatabaseDescriptor.getInvalidClientDisconnectJitter(TimeUnit.MILLISECONDS);
+        if (period < 0)
+        {
+            logger.info("Skipping invalid role disconnection");
+            return;
+        }
+        IntSupplier delayMillis = () -> period + ThreadLocalRandom.current().nextInt(0, jitter);
```

Comment:
This should be  a LongSupplier. In general, delay should be supplied as long, for consistency we should stick to long here as well.